src/handlers/gestures/gestureBuilder.ts

```diff
@@ -0,0 +1,113 @@
+import { SimpleGesture } from './simpleGestures';
+
+enum Relation {
+  Simultaneous,
+  Exclusive,
+  RequireToFail,
+}
+
+type PendingGesture = {
+  relation: Relation;
+  gesture: SimpleGesture;
+};
+
+export class GestureBuilder {
+  private pendingGestures: PendingGesture[] = [];
+
+  constructor(base: SimpleGesture) {
+    this.addGesture({ relation: Relation.Exclusive, gesture: base });
+  }
+
+  private addGesture(gesture: PendingGesture): GestureBuilder {
+    this.pendingGestures.push(gesture);
+    return this;
+  }
+
+  simultaneousWith(gesture: SimpleGesture): GestureBuilder {
+    return this.addGesture({
+      relation: Relation.Simultaneous,
+      gesture: gesture,
+    });
+  }
+
+  exclusiveWith(gesture: SimpleGesture): GestureBuilder {
+    return this.addGesture({
+      relation: Relation.Exclusive,
+      gesture: gesture,
+    });
+  }
+
+  requireToFail(gesture: SimpleGesture): GestureBuilder {
+    this.pendingGestures.push({
+      relation: Relation.RequireToFail,
+      gesture: gesture,
+    });
+
+    return this;
+  }
+
+  build(): BuiltGesture {
+    const result = new BuiltGesture(this.prepare);
+    result.gestures = this.pendingGestures.map((pending) => pending.gesture);
+    return result;
+  }
+
+  prepare = () => {
```

Comment:
This should be a normal method for consistency.

Reply:
I used arrow function there because it's passed as an argument and I needed `this` binding. Can I just change other methods to be arrow functions or bind it explicitly (and which one would be better), or is passing it a bad idea?

Reply:
Fixed in 233fcb5cb3a59975ca32c2620746cf70a413de5f and b07ee00debe2e5d8c84898e4cd5f6d83d37268a6